In [1]:
import pandas as pd
df=pd.read_csv("C:/professional course/pandas/Churn modeling - Copy/Churn modeling - Copy/Churn_Modelling.csv")
df.head()

RowNumber  CustomerId   Surname  CreditScore Geography  Gender  Age  \
0          1    15634602  Hargrave          619    France  Female   42   
1          2    15647311      Hill          608     Spain  Female   41   
2          3    15619304      Onio          502    France  Female   42   
3          4    15701354      Boni          699    France  Female   39   
4          5    15737888  Mitchell          850     Spain  Female   43   

   Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
0       2       0.00              1          1               1   
1       1   83807.86              1          0               1   
2       8  159660.80              3          1               0   
3       1       0.00              2          0               0   
4       2  125510.82              1          1               1   

   EstimatedSalary  Exited  
0        101348.88       1  
1        112542.58       0  
2        113931.57       1  
3         93826.63       0  
4         79084.10       0

In [3]:
churncustomers=df.groupby("Exited").get_group(1)
notchurncustomers=df.groupby("Exited").get_group(0)
cage=churncustomers["Age"].value_counts(ascending=False)
Gchurn=churncustomers.groupby("Geography").get_group("Germany")
Schurn=churncustomers.groupby("Geography").get_group("Spain")
Fchurn=churncustomers.groupby("Geography").get_group("France")
cfemale=churncustomers.groupby("Gender").get_group("Female")
nfemale=notchurncustomers.groupby("Gender").get_group("Female")
cmale=churncustomers.groupby("Gender").get_group("Male")
nmale=notchurncustomers.groupby("Gender").get_group("Male")
print(churncustomers.value_counts())
print(notchurncustomers.value_counts())






RowNumber  CustomerId  Surname     CreditScore  Geography  Gender  Age  Tenure  Balance    NumOfProducts  HasCrCard  IsActiveMember  EstimatedSalary  Exited
1          15634602    Hargrave    619          France     Female  42   2       0.00       1              1          1               101348.88        1         1
6525       15743293    Waters      651          Germany    Female  35   1       163700.78  3              1          1               29583.48         1         1
6582       15598744    Ch'ang      576          Germany    Female  71   6       140273.47  1              1          1               193135.25        1         1
6576       15651883    Genovesi    794          Germany    Female  55   6       115796.70  1              1          0               160526.36        1         1
6572       15747534    Torkelson   595          France     Male    46   10      0.00       1              1          0               73489.15         1         1
                                 

In [4]:
Schurn.head()

RowNumber  CustomerId    Surname  CreditScore Geography  Gender  Age  \
5           6    15574012        Chu          645     Spain    Male   44   
22         23    15699309  Gerasimov          510     Spain  Female   38   
30         31    15589475    Azikiwe          591     Spain  Female   39   
58         59    15623944      T'ien          511     Spain  Female   66   
86         87    15762418       Gant          750     Spain    Male   22   

    Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
5        8  113755.78              2          1               0   
22       4       0.00              1          1               0   
30       3       0.00              3          1               0   
58       4       0.00              1          1               0   
86       3  121681.82              1          1               0   

    EstimatedSalary  Exited  
5         149756.71       1  
22        118913.53       1  
30        140469.38       1  
58          1643.11       1  
86        128643.35       1

In [5]:
Fchurn.head()

RowNumber  CustomerId   Surname  CreditScore Geography  Gender  Age  \
0           1    15634602  Hargrave          619    France  Female   42   
2           3    15619304      Onio          502    France  Female   42   
35         36    15794171  Lombardo          475    France  Female   45   
41         42    15738148    Clarke          465    France  Female   51   
43         44    15755196    Lavine          834    France  Female   49   

    Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
0        2       0.00              1          1               1   
2        8  159660.80              3          1               0   
35       0  134264.04              1          1               0   
41       8  122522.32              1          0               0   
43       2  131394.56              1          0               0   

    EstimatedSalary  Exited  
0         101348.88       1  
2         113931.57       1  
35         27822.99       1  
41        181297.65       1  
43        194365.76       1

In [6]:
print(churncustomers["Geography"].value_counts())
print(notchurncustomers["Geography"].value_counts())

print(churncustomers["Gender"].value_counts())
print(notchurncustomers["Gender"].value_counts())


Geography
Germany    814
France     810
Spain      413
Name: count, dtype: int64
Geography
France     4204
Spain      2064
Germany    1695
Name: count, dtype: int64
Gender
Female    1139
Male       898
Name: count, dtype: int64
Gender
Male      4559
Female    3404
Name: count, dtype: int64


In [8]:
print("Churn customers of different country")
print("Germany\n")
print(Gchurn["Gender"].value_counts())
print("Spain\n")
print(Schurn["Gender"].value_counts())
print("France\n")
print(Fchurn["Gender"].value_counts())


Churn customers of different country
Germany

Gender
Female    448
Male      366
Name: count, dtype: int64
Spain

Gender
Female    231
Male      182
Name: count, dtype: int64
France

Gender
Female    460
Male      350
Name: count, dtype: int64


In [9]:
print(churncustomers["NumOfProducts"].value_counts())
print(notchurncustomers["NumOfProducts"].value_counts())

NumOfProducts
1    1409
2     348
3     220
4      60
Name: count, dtype: int64
NumOfProducts
2    4242
1    3675
3      46
Name: count, dtype: int64


In [12]:
print(notchurncustomers["Balance"].value_counts(ascending=False))
print(churncustomers["Balance"].value_counts(ascending=False))

Balance
0.00         3117
130170.82       2
122570.69       1
163085.79       1
197200.04       1
             ... 
138734.94       1
109819.16       1
176024.05       1
107499.70       1
130142.79       1
Name: count, Length: 4846, dtype: int64
Balance
0.00         500
129934.64      1
104339.11      1
142946.18      1
109908.06      1
            ... 
125647.36      1
138881.04      1
113715.36      1
81290.02       1
75075.31       1
Name: count, Length: 1538, dtype: int64


In [46]:

print(churncustomers["Balance"].median())
print(notchurncustomers["Balance"].median())
print(churncustomers["EstimatedSalary"].median())
print(notchurncustomers["EstimatedSalary"].median())

109349.29
92072.68
102460.84
99645.04


In [13]:
print(cfemale["Balance"].median())
print(nfemale["Balance"].median())
print(cmale["Balance"].median())
print(nmale["Balance"].median())

108076.33
89967.16500000001
111310.70999999999
93911.3


In [14]:
print(churncustomers["IsActiveMember"].value_counts())
print(notchurncustomers["IsActiveMember"].value_counts())

IsActiveMember
0    1302
1     735
Name: count, dtype: int64
IsActiveMember
1    4416
0    3547
Name: count, dtype: int64


Geography: If we divide the area of churn customers it can be seen that that France has the lower churn rate of  16.15% . And we will find the same in spain which is 16.67%. but in case of Germany the churnrate is twice then spain and france. so it can be assume that germany branch will not have a bright future without any modification.

Gender: First it can be seen that almost 20% of total customers are churned from bank. If we see in the gender distribution of churn customers we it can be observes that the churn rate is higher in female . Almost 55% of total churn customers are female.whether the male cursomers are less churn than female do.

Services: in the case of service has been taken most of the churn customer took 1 sevices almost 70% customer and the rest of the 30% took 2 or 3 sevices. but if we see those customers who are not churned most of them of 60% took 2 services.rest of them took 1 servise . So there is a point that those who chose one services there is a chance that they may churn in future.

BankBalace: There is a strange things found in the balance and estimated sallary of churn and not churn customers. the  average Bank balance and estimated salary is 5-6% higher in churncustomer than the not churn customers. Another shocking things is there about 3117 of 7963 of not churn customers which is about 39% customers have 0.00 bank balance. if we see in churn customers about 25% of people had 0.00 bank balance . so this may be hint that this 39% of not churn customer will churn in future.

Activity: like Bank balance here also noticable similler result which give the assumption of how many people are going to churn.
if we look down in churncustomers we can see that about 70% of churncustomers were inactive . so it indicate that 70% inactive not churncustomer which is about 2483 person can be churned in future.

in a nutshell the summmary of above is Germany branch are failing to hold their customer, the intension of churn are found most of in female customers, the customers who took only one sevice has possibility to churn, have 0.00 bank balance and inactive members can be churn in future.




